zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/UserTaskCorrectedApplierTest.java

```diff
@@ -0,0 +1,165 @@
+import java.util.List;
+import java.util.stream.Stream;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@ExtendWith(ProcessingStateExtension.class)
+public class UserTaskCorrectedApplierTest {
+
+  /** Injected by {@link ProcessingStateExtension} */
+  private MutableProcessingState processingState;
+
+  /** The class under test. */
+  private UserTaskCorrectedApplier userTaskCorrectedApplier;
+
+  /** Used for state assertions. */
+  private MutableUserTaskState userTaskState;
+
+  /** For setting up the state before testing the applier. */
+  private TestSetup testSetup;
+
+  @BeforeEach
+  public void setup() {
+    userTaskCorrectedApplier = new UserTaskCorrectedApplier(processingState);
+    userTaskState = processingState.getUserTaskState();
+    testSetup = new TestSetup(processingState);
+  }
+
+  /**
+   * Returns a stream of test case arguments. In order, these arguments are:
+   *
+   * <ul>
+   *   <li>the user task key
+   *   <li>a list of intents to apply to the state before applying the corrected event
+   *   <li>the expected lifecycle state after applying the corrected event
+   * </ul>
+   */
+  public static Stream<Arguments> testCases() {
+    return Stream.of(
+        Arguments.of(1, List.of(), LifecycleState.CREATING),
+        Arguments.of(
+            2, List.of(UserTaskIntent.CREATING, UserTaskIntent.CREATED), LifecycleState.ASSIGNING),
+        Arguments.of(
+            3, List.of(UserTaskIntent.CREATING, UserTaskIntent.CREATED), LifecycleState.UPDATING),
+        Arguments.of(
+            4,
+            List.of(UserTaskIntent.CREATING, UserTaskIntent.CREATED),
+            LifecycleState.COMPLETING));
+  }
+
+  @ParameterizedTest(name = "on {2}")
+  @MethodSource("testCases")
+  void shouldCorrectIntermediateUserTaskDataOnIntent(
+      final long userTaskKey, final List<UserTaskIntent> setup, final LifecycleState state) {
+    // given
+    final var given =
+        new UserTaskRecord()
+            .setUserTaskKey(userTaskKey)
+            .setAssignee("initial")
+            .setCandidateGroupsList(List.of("initial"))
+            .setCandidateUsersList(List.of("initial"))
+            .setDueDate("initial")
+            .setFollowUpDate("initial")
+            .setPriority(1);
+
+    setup.forEach(setupIntent -> testSetup.applyEventToState(userTaskKey, setupIntent, given));
+    testSetup.applyEventToState(userTaskKey, mapLifecycleStateToIntent(state), given);
+
+    Assumptions.assumeThat(userTaskState.getUserTask(userTaskKey)).isNotNull();
+    Assumptions.assumeThat(userTaskState.getLifecycleState(userTaskKey)).isEqualTo(state);
+    Assumptions.assumeThat(userTaskState.getIntermediateState(userTaskKey)).isNotNull();
```

Comment:
Oh, I didn’t realise that `AssertJ` also provides the [Assumptions](https://assertj.github.io/doc/#assertj-core-assumptions) concept - good to know!
I've previously used [assumeThat](https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html#assumeThat(T,%20org.hamcrest.Matcher)) from JUnit in some tests, but it’s not as convenient as the assumptions in `AssertJ`. TIL, thanks! 🙇